### PR TITLE
Add genai.protos

### PR DESCRIPTION
**Pull request by MarkDaoust**
The main changes are:

 - Add  `genai.protos` module, just containing `from google.ai.generativelanguage_v1beta.types import *`.
 - Replace all proto usage:
     - all `import google.ai.generativelanguage as glm` with `from google.generativeai import protos`.
     - all `glm.` with `protos.`.
 - Keep client usage as-is`glm.*Client`.
 - Add a test to police `glm.`
 - Fix `build_docs.py` script.

Manual changes: 

- docs/build_docs.py
- google/generativeai/protos.py
- tests/test_protos.py

### CODE REVIEW

File: tests/test_protos.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pathlib
+import re
+
+from absl.testing import parameterized
+import google.ai.generativelanguage as glm
+from google.generativeai import responder
```
[the remark this answers is not in the document]
Comment (MarkDaoust) [reply tests/test_protos.py:20]: Will fix all of these in another CL.

File: google/generativeai/protos.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+- ` including_default_value_fields:bool = True`, Set it to `False` to reduce the verbosity of the output.
+
+Additional arguments are described in the docstring:
+
+>>> help(proto.Part.to_dict)
+
+"""
+
+from google.ai.generativelanguage_v1beta.types import *
+from google.ai.generativelanguage_v1beta.types import __all__
```
[the remark this answers is not in the document]
Comment (MarkDaoust) [reply google/generativeai/protos.py:65]: I just want to set the __all__ as a hint to the doc-generator tool.

**Review by mayureshagashe2105 — APPROVED** [review]

File: google/generativeai/protos.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+The `genai` SDK tries to be permissive about what objects it will accept from a user, but in the end
+the SDK always converts input to an appropriate Proto Message object to send as the request.
+
+If you have any uncertainty about what the API may accept or return, these classes provide the
+complete/unambiguous answer. They come from the `google-ai-generativelanguage` package which is
+generated from a snapshot of the API definition.
+
+>>> from google.generativeai import protos
+>>> import inspect
+>>> print(inspect.getsource(protos.GenerateContentRequest))
```
Comment (markmcd) [inline google/generativeai/protos.py:29]: Any reason for choosing `GenerateContentRequest`? It's not exposed to the user in any way so might be confusing.

File: google/generativeai/protos.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+Proto classes can have "oneof" fields. Use `in` to check which `oneof` field is set.
+
+>>> p = protos.Part(text='hello')
+>>> 'text' in p
+True
+>>> p.inline_data = {'mime_type':'image/png', 'data': b'PNG'}
+>>> type(p.inline_data) is protos.Blob
+True
+>>> 'inline_data' in p
+False
```
Comment (markmcd) [inline google/generativeai/protos.py:40]: This should be True, and the next statement should be False

**Review by markmcd — DISMISSED** [review]
There's quite a lot of content that ends up in the protos/ dir - e.g. protos.GenerateContentRequest - that has no use in this SDK, so might be confusing.

The case I'm thinking of is a user typing some variation of `generate content` into the filter on the API page, and landing on protos.GenerateContentRequest/Response.

It also looks like the generated docs for protos.* still refer to `google.ai.generativelanguage` - I think that's fine, since it's both the truth and a dependency - but it means fields like `ToolConfig` are no longer linked.

LGTM anyway as this is an improvement, and maybe doesn't matter with API ref changes coming.

File: tests/test_protos.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import re
+
+from absl.testing import parameterized
+
+ROOT = pathlib.Path(__file__).parent.parent
+
+
+class UnitTests(parameterized.TestCase):
+    def test_check_glm_imports(self):
+        for fpath in ROOT.rglob("*.py"):
```
[the remark this answers is not in the document]
Comment (markmcd) [reply tests/test_protos.py:25]: Ehhhh don't mind me - there's too many dirs to list explicitly so I can store my junk elsewhere :)

File: google/generativeai/protos.py (hunk 3 above)
Comment (MarkDaoust) [reply google/generativeai/protos.py:29]: Replaced with `Part`.

File: google/generativeai/protos.py (hunk 4 above)
Comment (MarkDaoust) [reply google/generativeai/protos.py:40]: Right, thanks.

**Comment by MarkDaoust** [comment]
> protos.GenerateContentRequest - that has no use in this SDK, so might be confusing

Yes, it just didn't seem worth sorting through these 1 by one to decide which to include and which not to include.

> still refer to google.ai.generativelanguage - I think that's fine, since it's both the truth and a dependency - but it means fields like ToolConfig are no longer linked.

Is there any way we can fix this? Add a find and replace in the generation process?

> and maybe doesn't matter with API ref changes coming.

Yeah, that turns it into a real API-reference. But leaves us with nowhere that the SDKs are described in detail?

**Review by markmcd — APPROVED** [review]

**Pull request merged**